### ISSUE TITLE
Support needed for Terragrunt Optional Dependencies

### ISSUE THREAD

**Issue by bdorplatt** [infracost/infracost#2706]
Infracost on the latest version is not able to process the new "optional dependencies" feature of Terragrunt released in version [v0.50.17](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.50.17)
- Added enabled property on dependency block, once set to false - dependency will be skipped.

If the enabled flag is used in a Terragrunt dependency in a terragrunt.hcl file, it results in the following error:

`Unsupported argument; An argument named "enabled" is not expected here.`

Here is an example of how the new feature is used:

```
dependency "sql_server" {
  enabled = local.environment_vars.locals.deploy_sql == "yes" ? true : false
  config_path = "../../sqlserver/sqlserver01"
  mock_outputs = {
    sql_server_id   = "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/mock-rg/providers/Microsoft.Sql/servers/mock-sql"
    sql_server_fqdn = "bdous-nc-devtest-mock-sql.database.net"
    sql_server_name = "mocksqlname"
  }
  mock_outputs_allowed_terraform_commands = ["validate", "plan"]
  mock_outputs_merge_strategy_with_state  = "shallow"
}
```

**Comment by aliscott** [comment]
Thanks for reporting @bdorplatt 🙏. We will look at getting a fix out for the next release.

**Comment by aliscott** [comment]
@bdorplatt this is fixed in master now as a result of https://github.com/infracost/infracost/pull/2708. This will go out in the next release (scheduled for next week).

**Comment by aliscott** [comment]
@bdorplatt, @LoganHealy, @kbargobdo, and @lpmurphybdo this is now released in [v0.10.30](https://github.com/infracost/infracost/releases/tag/v0.10.30).
